File: app/src/main/java/com/islam/basepropject/ui/ExPaging/ExFragment.java
package com.islam.basepropject.ui.ExPaging;

import android.os.Bundle;
import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

import com.islam.basepropject.R;
import com.islam.basepropject.project_base.base.fragments.BaseSuperFragment;

public class ExFragment extends BaseSuperFragment<ExViewModel> {

    private ExAdapter mAdapter;

    @Override
    protected void onLaunch() {
        initContentView(R.layout.fragment_fragment2);
        initToolbar(0, false);
        initViewModel(this, ExViewModel.class);
    }


    @Override
    protected void onViewCreated(View view, ExViewModel viewModel, Bundle instance) {
        mAdapter = new ExAdapter();

        createRecyclerView(mAdapter);
    }

    @Override
    protected void setUpObservers() {
        mViewModel.getOrders().subscribe(strings -> mAdapter.submitList(strings));

        mViewModel.getNetworkState().observe(getViewLifecycleOwner(),
                networkRequestState -> mAdapter.setNetworkState(networkRequestState));
    }

}

File: app/src/main/java/com/islam/basepropject/ui/ExPaging/ExFragment.kt
package com.islam.basepropject.ui.ExPaging

import android.os.Bundle
import android.view.View
import androidx.lifecycle.Observer

import com.islam.basepropject.R
import com.islam.basepropject.project_base.base.fragments.BaseSuperFragment

class ExFragment : BaseSuperFragment<ExViewModel>() {
    override var fragmentTag = "ExFragment"
    private var mAdapter = ExAdapter()

    override fun onLaunch() {
        initContentView(R.layout.fragment_fragment2)
        initToolbar(0, false)
        initViewModel(this, ExViewModel::class.java)
    }


    override fun onViewCreated(view: View, viewModel: ExViewModel?, instance: Bundle?) {
        mViewModel?.loadOrders(createRecyclerView(mAdapter).id)
    }

    override fun setUpObservers() {
        mViewModel!!.orders?.observe(viewLifecycleOwner, Observer {mAdapter.submitList(it)})
    }

}
